src/adapters/rival-bears-staking.adapter.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright Abridged, Inc. 2022. All Rights Reserved.
+// Node module: @collabland/staking-contracts
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking';
+import type {RivalBearsStaking} from '../types/RivalBearsStaking';
+// Use the full path to import instead of `../types`
+import {RivalBearsStaking__factory} from '../types/factories/RivalBearsStaking__factory';
+
+@injectable(
+  {
+    scope: BindingScope.SINGLETON, // Mark the adapter as a singleton
+  },
+  // Mark it as an extension to staking contracts service
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class RivalBearsStakingContractAdapter extends BaseStakingContractAdapter {
+  private contract: RivalBearsStaking;
+  /**
+   * The contract address
+   */
+  contractAddress = '0xd1A53bFbFF6E36b1E1436b2029582e6Bc86E2816';
```

Comment:
This address is not a deployed contract on ploygon - https://polygonscan.com/address/0xd1A53bFbFF6E36b1E1436b2029582e6Bc86E2816